lychee-lib/src/utils/fragment_checker.rs

```diff
@@ -59,19 +59,20 @@ impl FragmentChecker {
         if file_type == FileType::Markdown {
             fragment_decoded = fragment_decoded.to_lowercase().into();
         }
+        // Empty # and #top fragments are always valid, triggering the browser to scroll to top.
+        let is_emtpty_or_top_fragment = fragment.is_empty() || fragment.eq_ignore_ascii_case("top");
         match self.cache.lock().await.entry(url_without_frag) {
             Entry::Vacant(entry) => {
                 let content = fs::read_to_string(path).await?;
                 let file_frags = extractor(&content);
                 let contains_fragment =
                     file_frags.contains(fragment) || file_frags.contains(&fragment_decoded as &str);
                 entry.insert(file_frags);
-                Ok(contains_fragment)
-            }
-            Entry::Occupied(entry) => {
-                Ok(entry.get().contains(fragment)
-                    || entry.get().contains(&fragment_decoded as &str))
+                Ok(is_emtpty_or_top_fragment || contains_fragment)
```

Comment:
Probably Rust optimises such automatically, but in code logic, it does not make sense to check first whether the fragment is contained, if it is empty or `top`. So instead of creating `contains_fragment`, it could be better in terms of performance/logic to align it with below, to skip `contains_fragment =` and instead do:
```suggestion
                Ok(is_emtpty_or_top_fragment
                    || file_frags.contains(fragment)
                    || file_frags.contains(&fragment_decoded as &str))
```

Reply:
Ah no, that does not work since `file_frags` is moved by `insert`. Another approach, insert it if missing, and check the cache entry in any case afterwards.

Reply:
Now it is checked and returned much earlier, if the fragment is empty or "top". No point to further process the fragment and URL, create or check the cache entry etc. Those cases can be handled just the same way as if there was no fragment at all.

Reply:
@MichaIng 
yes, indeed - this is a much better option (I'll start to to look from these perspectives too - thank you!)

one quick observation - by exiting earlier, we're postponing the extraction operation (to later) and thereby the cache updates - I don't see it creating any issues but wanted to note it here.

Reply:
Yeah I was also thinking about the skipped/delayed cache entry, but it actually makes sense:
- Either the same URL is not checked with any other fragment later, then the cache entry is now skipped. It would have remained unused anyway.
- Or the same URL is checked with another fragment later, then the cache entry is created at this later point, which does not cause any downside.